### ISSUE TITLE
cm001 is writting to an invalid address

### ISSUE THREAD

**Issue by ubitux** [JCWasmx86/Crackme#2]
`movupd %xmm1, (tmp)` ← this is incorrect because `tmp` ends up being located at the end of the `.bss` section.

Basically in the C, this:
```c
uint64_t tmp;
uint64_t tmp2;
char* pwd;
```
Creates an inverted layout:
![2022-10-24-230751-Meisoh4i](https://user-images.githubusercontent.com/34467/197629740-18ffb570-6f7d-4054-a017-4aca78a755f4.png)

So after the 64-bit long `tmp` there is nothing: it's the end of the section, the following address is supposedly unmapped. Writing into it is actually overwriting into the memory (and since you're actually reading from there afterwards, it's even more an undefined behavior).

I don't think the compiler can provide any guarantee about how these fields are going to be layed out. Maybe wrapping them into a structure, or simply creating a single `uint8_t tmp[16]` would do the trick just fine.

I'm surprised this doesn't crash right away, so feel free to correct me if I'm missing something. It even actually seems to work for some reason, so I have no idea where the data is actually being written.

Even valgrind doesn't seem to cause a ruckus about it (probably a rare scenario). Note that the program doesn't behave well with valgrind due to `RDX` being uninitialized, but I think you're already aware of that.

**Comment by JCWasmx86** [comment]
Yes, cm001 is sadly buggy beyond salvation. But I don't have time to fix that and won't accept fixes as I already uploaded the binary to crackmes.one a long time ago and want to preserve the exact source code I used

But thanks for your investigation.

Edit) Or well I would accept changes, if the binaries are bit-for-bit identical (Except stuff like the .note.gnu* sections, .comment and so on and really necessary changes to the other section)
